File: src/jvmMain/java/org/hisp/dhis/lib/expression/eval/ValueTypeVariableValue.java
package org.hisp.dhis.lib.expression.eval;

import lombok.Value;
import org.hisp.dhis.lib.expression.spi.ValueType;
import org.hisp.dhis.lib.expression.spi.VariableValue;

import java.util.List;

@Value
public class ValueTypeVariableValue implements VariableValue {

    ValueType valueType;

    @Override
    public String value() {
        return null;
    }

    @Override
    public Object valueOrDefault() {
        return null;
    }

    @Override
    public List<String> candidates() {
        return null;
    }

    @Override
    public String eventDate() {
        return null;
    }

    @Override
    public ValueType valueType() {
        return valueType;
    }
}

File: src/jvmMain/java/org/hisp/dhis/lib/expression/eval/ValueTypeVariableValue.kt
package org.hisp.dhis.lib.expression.eval

import org.hisp.dhis.lib.expression.spi.ValueType
import org.hisp.dhis.lib.expression.spi.VariableValue

class ValueTypeVariableValue(private val valueType: ValueType) : VariableValue {

    override fun value(): String? {
        return null
    }

    override fun valueOrDefault(): Any? {
        return null
    }

    override fun candidates(): List<String?>? {
        return null
    }

    override fun eventDate(): String? {
        return null
    }

    override fun valueType(): ValueType {
        return valueType
    }
}
